dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreRecordCollection.cs

```diff
@@ -407,7 +407,7 @@ public Task<VectorSearchResults<TRecord>> HybridSearchAsync<TVector>(TVector vec
         this.VerifyVectorType(vector);
 
         var searchOptions = options ?? s_defaultKeywordVectorizedHybridSearchOptions;
-        var vectorProperty = this._propertyReader.GetVectorPropertyOrSingle(searchOptions.VectorPropertyName);
+        var vectorProperty = this._propertyReader.GetVectorPropertyOrSingle<TRecord>(new() { VectorProperty = searchOptions.VectorProperty });
```

Comment:
When we remove the `[Obsolete] VectorPropertyName` property, the `GetVectorPropertyOrSingle` is going to accept just `Expression<Func<TRecord, object?>>?` and there will be no need to instantiate the search option class here.

(I did not want to send a string and expression, as I would need to touch the obsolete property in every connector, and pragmas to disable it and simply require more changes now and in the future)